src/main/java/me/botsko/prism/actions/BlockAction.java

```diff
@@ -350,50 +350,52 @@ ChangeResult placeBlock(Player player, QueryParameters parameters, boolean isPre
         BlockActionData blockActionData = getActionData();
+        if(blockActionData != null ) {
+            if ((getMaterial() == PLAYER_HEAD || getMaterial() == PLAYER_WALL_HEAD)
+                    && blockActionData instanceof SkullActionData) {
+                return handleSkulls(block, blockActionData, originalBlock);
+            }
 
-        if ((getMaterial() == PLAYER_HEAD || getMaterial() == PLAYER_WALL_HEAD)
-                && blockActionData instanceof SkullActionData) {
-            return handleSkulls(block, blockActionData, originalBlock);
-        }
-
-        if (getMaterial() == SPAWNER && blockActionData instanceof SpawnerActionData) {
+            if (getMaterial() == SPAWNER && blockActionData instanceof SpawnerActionData) {
 
-            final SpawnerActionData s = (SpawnerActionData) blockActionData;
+                final SpawnerActionData s = (SpawnerActionData) blockActionData;
 
-            // Set spawner data
-            ((CreatureSpawner) newState).setDelay(s.getDelay());
-            ((CreatureSpawner) newState).setSpawnedType(s.getEntityType());
+                // Set spawner data
+                ((CreatureSpawner) newState).setDelay(s.getDelay());
+                ((CreatureSpawner) newState).setSpawnedType(s.getEntityType());
 
-        }
+            }
 
-        if (getMaterial() == COMMAND_BLOCK
-                && blockActionData instanceof CommandActionData) {
-            final CommandActionData c = (CommandActionData) blockActionData;
-            ((CommandBlock) newState).setCommand(c.command);
-        }
-        if (newState instanceof Nameable && actionData.customName != null) {
-            ((Nameable) newState).setCustomName(actionData.customName);
-        }
-        if (parameters.getProcessType() == PrismProcessType.ROLLBACK
-                && Tag.SIGNS.isTagged(getMaterial())
-                && blockActionData instanceof SignActionData) {
-
-            final SignActionData s = (SignActionData) blockActionData;
-            // Verify block is sign. Rarely, if the block somehow pops off
-            // or fails
-            // to set it causes ClassCastException:
-            // org.bukkit.craftbukkit.v1_4_R1.block.CraftBlockState
-            // cannot be cast to org.bukkit.block.Sign
-            // https://snowy-evening.com/botsko/prism/455/
-            if (newState instanceof Sign) {
-                if (s.lines != null) {
-                    for (int i = 0; i < s.lines.length; ++i) {
-                        ((Sign) newState).setLine(i, s.lines[i]);
+            if (getMaterial() == COMMAND_BLOCK
+                    && blockActionData instanceof CommandActionData) {
+                final CommandActionData c = (CommandActionData) blockActionData;
+                ((CommandBlock) newState).setCommand(c.command);
+            }
+            if (newState instanceof Nameable && blockActionData.customName != null && !blockActionData.customName.equals("")) {
```

Comment:
Codacy found an issue: [Line is longer than 120 characters (found 127).](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5911179)